firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -215,34 +215,40 @@ public UpdateTask updateIfNewReleaseAvailable() {
       showSignInDialogTask = new TaskCompletionSource<>();
     }
 
-    signInConfirmationDialog = new AlertDialog.Builder(hostActivity).create();
     dialogHostActivity = hostActivity;
 
-    Context context = firebaseApp.getApplicationContext();
-    signInConfirmationDialog.setTitle(context.getString(R.string.signin_dialog_title));
-    signInConfirmationDialog.setMessage(context.getString(R.string.singin_dialog_message));
-
-    signInConfirmationDialog.setButton(
-        AlertDialog.BUTTON_POSITIVE,
-        context.getString(R.string.singin_yes_button),
-        (dialogInterface, i) -> showSignInDialogTask.setResult(null));
-
-    signInConfirmationDialog.setButton(
-        AlertDialog.BUTTON_NEGATIVE,
-        context.getString(R.string.singin_no_button),
-        (dialogInterface, i) ->
-            showSignInDialogTask.setException(
-                new FirebaseAppDistributionException(
-                    ErrorMessages.AUTHENTICATION_CANCELED, AUTHENTICATION_CANCELED)));
-
-    signInConfirmationDialog.setOnCancelListener(
-        dialogInterface ->
-            showSignInDialogTask.setException(
-                new FirebaseAppDistributionException(
-                    ErrorMessages.AUTHENTICATION_CANCELED, AUTHENTICATION_CANCELED)));
-
-    signInConfirmationDialog.show();
-
+    // We may not be on the main (UI) thread in some cases, specifically if the developer calls
+    // the basic config from the background. If we are already on the main thread, this will
+    // execute immediately.
+    hostActivity.runOnUiThread(
```

Comment:
Can we do the work of creating the dialog not on the UI thread? Can we just have the showing of the dialog on the UI thread?

Reply:
Good question - we actually can't. `new AlertDialog.Builder(hostActivity).create()` has to be called on the UI thread. I at first just wrapped the `.show()` call and then had to figure out why it was still broken :)